contracts/leaser/src/state/leases.rs

```diff
@@ -74,40 +74,18 @@ impl Leases {
         })
     }
 
-    pub fn iter(storage: &dyn Storage) -> impl Iterator<Item = ContractResult<Addr>> + '_ {
+    pub fn iter(
+        storage: &dyn Storage,
+        next_customer: Option<Addr>,
+    ) -> impl Iterator<Item = ContractResult<Customer<IntoIter<Addr>>>> + '_ {
```

Comment:
(Note) Please, refrain from using `impl Trait` in return position unless absolutely necessary until it is stabilized in trait methods too. It becomes a pain to refactor later.

Reply:
what's the relation to the unstable impl in Trait methods? Using concrete types has always been a bigger pain.

Reply:
The relation is the refactoring of the `timealarms` contract that I did a few days ago, because I needed to extract one method into a trait and thus opened Pandora's box. I had to go through several layers of return types to be able to deal with it.